test/python/algorithms/test_amplitude_estimators.py

```diff
@@ -0,0 +1,513 @@
+
+    def __init__(self, probability):
+        super().__init__(1, global_phase=np.pi)
+        self.angle = 2 * np.arcsin(np.sqrt(probability))
+        self.ry(2 * self.angle, 0)
+
+        # # oracle
+        # self.x(0)
+        # self.z(0)
+        # self.x(0)
+
+        # # A dagger
+        # self.ry(-self.angle, 0)
+
+        # # zero reflection
+        # self.x(0)
+        # self.z(0)
+        # self.x(0)
+
+        # # A
+        # self.ry(self.angle, 0)
+
+    def power(self, power, matrix_power=False):
+        if matrix_power:
+            return super().power(power, True)
+
+        powered = QuantumCircuit(1)
+        powered.ry(power * 2 * self.angle, 0)
+        return powered
+
+
+class SineIntegral(QuantumCircuit):
+    r"""Construct the A operator to approximate the integral
+
+        \int_0^1 \sin^2(x) d x
+
+    with a specified number of qubits.
+    """
+
+    def __init__(self, num_qubits):
+        qr_state = QuantumRegister(num_qubits, 'state')
+        qr_objective = QuantumRegister(1, 'obj')
+        super().__init__(qr_state, qr_objective)
+
+        # prepare 1/sqrt{2^n} sum_x |x>_n
+        self.h(qr_state)
+
+        # apply the sine/cosine term
+        self.ry(2 * 1 / 2 / 2 ** num_qubits, qr_objective[0])
+        for i, qubit in enumerate(qr_state):
+            self.cry(2 * 2 ** i / 2 ** num_qubits, qubit, qr_objective[0])
+
+
+@ddt
+class TestBernoulli(QiskitAlgorithmsTestCase):
+    """Tests based on the Bernoulli A operator.
+
+    This class tests
+        * the estimation result
+        * the constructed circuits
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self._statevector = QuantumInstance(backend=BasicAer.get_backend('statevector_simulator'),
+                                            seed_simulator=2, seed_transpiler=2)
+        self._unitary = QuantumInstance(backend=BasicAer.get_backend('unitary_simulator'), shots=1,
+                                        seed_simulator=42, seed_transpiler=91)
+
+        def qasm(shots=100):
+            return QuantumInstance(backend=BasicAer.get_backend('qasm_simulator'), shots=shots,
+                                   seed_simulator=2, seed_transpiler=2)
+        self._qasm = qasm
+
+    @idata([
+        [0.2, AmplitudeEstimation(2), {'estimation': 0.5, 'mle': 0.2}],
+        [0.49, AmplitudeEstimation(3), {'estimation': 0.5, 'mle': 0.49}],
+        [0.2, MaximumLikelihoodAmplitudeEstimation(2), {'estimation': 0.2}],
+        [0.49, MaximumLikelihoodAmplitudeEstimation(3), {'estimation': 0.49}],
+        [0.2, IterativeAmplitudeEstimation(0.1, 0.1), {'estimation': 0.2}],
+        [0.49, IterativeAmplitudeEstimation(0.001, 0.01), {'estimation': 0.49}],
+        [0.2, FasterAmplitudeEstimation(0.1, 3, rescale=False), {'estimation': 0.2}],
+        [0.12, FasterAmplitudeEstimation(0.1, 2, rescale=False), {'estimation': 0.12}]
+    ])
+    @unpack
+    def test_statevector(self, prob, qae, expect):
```

Comment:
Maybe something to think about. It might have been nicer to have had some base class for common methods like this so there could be one test case file per AE algorithm with just the data and call up to super method needed in the derived class and any algo specific tests.